### PR TITLE
Feature/248233 - Shortcuts for document splitting

### PR DESCRIPTION
**Pull request by Chris2011**
This PR will fix this original ticket: https://netbeans.org/bugzilla/show_bug.cgi?id=248233

* Created action for horizontally split and added shortcut
* Created action for vertically split and added shortcut
* Created action for clear split and added shortcut
* Refactored code a bit.

Regards

Chris

### CODE REVIEW

**Comment by Chris2011** [comment]
Created a JIRA ticket too: https://issues.apache.org/jira/browse/NETBEANS-59

**Comment by tusharvjoshi** [comment]
I verified the steps provided in JIRA ticket as follows:
1. Open any document what you want
2. Hit ctrl + shift + h to split the document horizontally
3. Hit ctrl + shift + v to split the document vetically
4. Hit ctrl + shift + c to clear split the document
5. Play around of hitting randomly the shortcuts

As I am working on macOS the shortcuts I had to use were comm + ctrl + shift + h, comm + ctrl + shift + v and comm + ctrl + shift + c

I confirm the working of these shortcuts as explained in the steps.  Usually when we need keyboard shortcuts we also add Menu options for the actions, can we apply menu options for these actions also?

**Comment by jlahoda** [comment]
Chris, as was discussed elsewhere, the patches change line endings to Windows line endings (CRLF), and so look like deletion and addition of each line in the file (including the license headers, which I suppose you didn't want to change). Please fix this, ensuring the patch is using the just "LF" as line endings, and is only changing what should/needs to be changed. This is to keep the history clean, and to aid reviews. I'd also suggest to squash the commits and rebase, so that there's only one commit merged when this is being merged.

Thanks.

**Comment by Chris2011** [comment]
So what should I do exactly here? Changing the line endings back or what? It was not my intention to change that of course not. But I created 3 new files so all of these are really new files, which didn't exist before. Only to let you know that.

**Comment by junichi11** [comment]
I've confirmed that your PR files have windows line endings.

Please try to check your line endings using the following plugin:
http://plugins.netbeans.org/plugin/36810/show-and-change-line-endings

The cause may be `core.autocrlf` configuration. Please try to check it:
```
git config -l
```
When you commit your changes, if line endings are not changed to `LF`, please try to change it to `false`.
e.g.
```
git config --global core.autocrlf false
```
Then, clone the repository again.

One Note:
If you create a new file in windows, its line ending is `CRLF` by default. So you have to change it.

Thanks.

**Comment by Chris2011** [comment]
@junichi11 thx for the info, yes I confirmed the line endings of CRLF too. So I changed the autocrlf from true to false. So now I have to clone it again? Maybe not possible to create a new feature branch and change the line endings?

So as I understand it right, I have to make a new PR with the changes, that I made again?

**Comment by junichi11** [comment]
> So now I have to clone it again?

Perhaps, need not do it if the other file line endings are not `CRLF`. If you want to use this PR, you can do the following steps (I can be wrong):
1. Remove your remote branch once
2. Fix your line endings for your changes. then commit
3. To create just one commit, rebase (or git reset --soft and commit?) 
4. Push your branch again

BTW, Did you read an email[1] from Jan? Probably, you should wait a while. 

[1] https://lists.apache.org/thread.html/ed2718660a4c6a961be15d5d6c9febec9fc19b5ca5fb0d81eab34f0c@%3Cdev.netbeans.apache.org%3E

**Comment by Chris2011** [comment]
Yes, I knew that mail. But there is no official comment about that and no decision made yet.

**Comment by Chris2011** [comment]
See this PR #3

File: core.multiview/src/org/netbeans/core/multiview/actions/SplitDocumentVerticallyAction.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * @author Chris2011
+ */
+@ActionID(
+    category = "Tools",
+    id = "org.netbeans.core.multiview.SplitDocumentVerticallyAction"
+)
+@ActionRegistration(
+    displayName = "#LBL_ValueSplitVertical"
+)
+@ActionReference(path = "Shortcuts", name = "DOS-V")
```
Comment (Chris2011) [inline core.multiview/src/org/netbeans/core/multiview/actions/SplitDocumentVerticallyAction.java]: Added shortcut

**Pull request closed without merging**